modules/woocommerce-analytics/classes/wp-woocommerce-analytics-universal.php

```diff
@@ -223,22 +237,14 @@ public function checkout_process() {
 				continue;
 			}
 
-			$product_details = $this->get_product_details( $product );
-
-			$universal_commands[] = "_wca.push( {
-				'_en': 'woocommerceanalytics_product_checkout',
-				'blog_id': '" . esc_js( $blogid ) . "',
-				'pi': '" . esc_js( $product_details['id'] ) . "',
-				'pn': '" . esc_js( $product_details['name'] ) . "',
-				'pc': '" . esc_js( $product_details['category'] ) . "',
-				'pp': '" . esc_js( $product_details['price'] ) . "',
-				'pq': '" . esc_js( $cart_item['quantity'] ) . "',
-				'pt': '" . esc_js( $product_details['type'] ) . "',
-				'ui': '" . esc_js( $this->get_user_id() ) . "',
-			} );";
+			$this->record_event(
+				'woocommerceanalytics_product_checkout',
+				$product->get_id(),
+				array(
+					'pq' => $cart_item['quantity'],
+				)
+			);
 		}
-
-		wc_enqueue_js( implode( "\r\n", $universal_commands ) );
```

Comment:
Do we need to include this in `record_event`?

Reply:
I don't think so - this was just rendering multiple `_wca.push()` calls (as JS strings) with whitespace in between.

In `record_event` we're enqueuing the `_wca.push()` call directly. 

The way to test this is to test an event that trigger on page load; for example add-to-cart from shop page, then view product page.